File: server/src/main/java/io/edurt/datacap/server/body/ExecuteDslBody.java
package io.edurt.datacap.server.body;

import io.edurt.datacap.plugin.builder.sql.configure.SqlBody;
import io.edurt.datacap.spi.FormatType;
import lombok.Data;
import lombok.ToString;

import java.util.Map;

@Data
@ToString
public class ExecuteDslBody
{
    private String name;
    private SqlBody configure;
    private Map<String, Object> env;
    private FormatType format = FormatType.NONE;
}

File: core/datacap-web/src/model/ExecuteDslBody.ts
import {SqlBody} from "@/model/builder/SqlBody";

export class ExecuteDslBody
{
  name: string;
  configure: SqlBody;
  env?: object;
  format?: string;

  constructor(builder: ExecuteDslBodyBuilder)
  {
    this.name = builder.name;
    this.configure = builder.configure;
    this.format = builder.format;
  }
}

export class ExecuteDslBodyBuilder
{
  private readonly _name: string;
  private readonly _format: string;
  private _configure: SqlBody;

  constructor(name: string, format: string)
  {
    this._name = name;
    this._format = format;
  }

  setConfigure(configure: SqlBody)
  {
    this._configure = configure;
    return this;
  }

  get name()
  {
    return this._name;
  }

  get configure()
  {
    return this._configure;
  }

  get format()
  {
    return this._format;
  }

  build()
  {
    return new ExecuteDslBody(this);
  }
}
